src/openms_gui/source/VISUAL/APPLICATIONS/TOPPViewBase.cpp

```diff
@@ -614,6 +620,49 @@ namespace OpenMS
           throw Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No peptide identifications with sufficient information remaining.");
         }
         peptides.swap(peptides_with_rt);
+
+        if (!proteins.empty())
+        {
+          StringList paths;
+          proteins[0].getPrimaryMSRunPath(paths);
+
+          for (StringList::const_iterator it = paths.begin(); it != paths.end(); ++it)
+          {
+
+            if (File::exists(*it) && fh.getType(*it) == FileTypes::MZML)
+            {
+              annotate_data.path = *it;
+            }
+          }
+          // annotation could not be found in file reference
+          if (annotate_data.path.empty())
+          {
+            // try to find file with same path & name but with mzML extension
+            auto target = fh.swapExtension(abs_filename, FileTypes::Type::MZML);
+            if (File::exists(target)) {
+              annotate_data.path = target;
+            }
+          }
+
+          if (!annotate_data.path.empty())
+          {
+            // open dialog for annotation on load
+            QMessageBox msg_box;
+            auto name = File::basename(annotate_data.path);
+            msg_box.setText("An associated Spectra file was found.");
+            msg_box.setInformativeText(String("Annotate spectra using " + name + "?").toQString());
```

Comment:
is name here the spectra file or the id file?

Reply:
Thats the spectra file. I'll rename the variable to be more descriptive.

Reply:
because then the the text does not match that well

Reply:
Right. Would something like: "Annotate spectra in _name_?" be better?